plugin/src/test/groovy/com/novoda/staticanalysis/internal/findbugs/FindbugsIntegrationTest.groovy

```diff
@@ -202,4 +230,129 @@ class FindbugsIntegrationTest {
+                }''')
+                .build('check')
+
+        Truth.assertThat(result.outcome(':findbugsDebugAndroidTest')).isEqualTo(TaskOutcome.UP_TO_DATE)
+        Truth.assertThat(result.outcome(':generateFindbugsDebugAndroidTestHtmlReport')).isEqualTo(TaskOutcome.SKIPPED)
+        Truth.assertThat(result.outcome(':findbugsDebug')).isEqualTo(TaskOutcome.SUCCESS)
+        Truth.assertThat(result.outcome(':generateFindbugsDebugHtmlReport')).isEqualTo(TaskOutcome.SUCCESS)
+        Truth.assertThat(result.outcome(':findbugsDebugUnitTest')).isEqualTo(TaskOutcome.UP_TO_DATE)
+        Truth.assertThat(result.outcome(':generateFindbugsDebugUnitTestHtmlReport')).isEqualTo(TaskOutcome.SKIPPED)
+        Truth.assertThat(result.outcome(':findbugsRelease')).isEqualTo(TaskOutcome.UP_TO_DATE)
+        Truth.assertThat(result.outcome(':generateFindbugsReleaseHtmlReport')).isEqualTo(TaskOutcome.SKIPPED)
+    }
+
+    @Test
+    public void shouldProvideNoClassesToFindbugsTaskWhenNoJavaSourcesToAnalyse() {
+        TestProject project = projectRule.newProject()
+        assumeThat(project).isJavaProject()
+
+        TestProject.Result result = project
+                .withSourceSet('debug', SOURCES_WITH_LOW_VIOLATION, SOURCES_WITH_MEDIUM_VIOLATION)
+                .withSourceSet('test', SOURCES_WITH_HIGH_VIOLATION)
+                .withPenalty('''{
+                    maxErrors = 0
+                    maxWarnings = 10
+                }''')
+                .withFindbugs('findbugs { exclude project.sourceSets.test.java.srcDirs }')
+                .withAdditionalConfiguration(addCheckFindbugsClassesTask())
+                .build('checkFindbugsClasses')
+
+        Truth.assertThat(result.outcome(':checkFindbugsClasses')).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    public void shouldProvideNoClassesToFindbugsTaskWhenNoAndroidSourcesToAnalyse() {
+        TestProject project = projectRule.newProject()
+        assumeThat(project).isAndroidProject()
+
+        TestProject.Result result = project
+                .withSourceSet('debug', SOURCES_WITH_LOW_VIOLATION)
+                .withSourceSet('test', SOURCES_WITH_HIGH_VIOLATION)
+                .withSourceSet('androidTest', SOURCES_WITH_HIGH_VIOLATION)
+                .withPenalty('''{
+                    maxErrors = 0
+                    maxWarnings = 1
+                }''')
+                .withFindbugs('''findbugs {
+                    exclude project.android.sourceSets.test.java.srcDirs
+                    exclude project.android.sourceSets.androidTest.java.srcDirs
+                }''')
+                .withAdditionalConfiguration(addCheckFindbugsClassesTask())
+                .build('checkFindbugsClasses')
+
+        Truth.assertThat(result.outcome(':checkFindbugsClasses')).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    /**
+     * <p>While integrating the plugin in one of our projects we realised that Findbugs was analysing a source folder
```

Comment:
No, I don't think it adds more insight really. I wanted to put some explanation on why that (gross) task exists tho, maybe I can shave all the context and leave only the hint about the check?